bbq/merge_schema.go

```diff
@@ -0,0 +1,36 @@
+package bbq
+
+import "cloud.google.com/go/bigquery"
+
+func mergeFieldsToBqSchema(existingSchema, newSchema bigquery.Schema) bigquery.Schema {
+	for _, fieldSchema := range newSchema {
+		if !isNameExists(existingSchema, fieldSchema.Name) {
+			existingSchema = append(existingSchema, fieldSchema)
+			continue
+		}
+
+		getChildSchema := getSchemaByName(existingSchema, fieldSchema.Name)
+		if getChildSchema != nil {
+			getChildSchema.Schema = mergeFieldsToBqSchema(getChildSchema.Schema, fieldSchema.Schema)
+		}
```

Comment:
maybe we can call this just `childField` because `get` sounds like a function. And it's not a schema, it's a field. childField.Schema is the schema

Reply:
or `matchedChildField`